src/api/libcellml/printer.h

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+
+#include "libcellml/enumerations.h"
+#include "libcellml/logger.h"
+#include "libcellml/types.h"
+
+//! Everything in libCellML is in this namespace.
+namespace libcellml {
+
+/**
+ * @brief The Printer class.
+ *
+ * The Printer class is for representing a CellML Printer.
+ */
+class LIBCELLML_EXPORT Printer: public Logger
+{
+public:
+    Printer(Format format); /**< Constructor */
+    ~Printer(); /**< Destructor */
+    Printer(const Printer &rhs); /**< Copy constructor */
+    Printer(Printer &&rhs); /**< Move constructor */
+    Printer& operator=(Printer p); /**< Assignment operator */
+
+    /**
+     * @brief Serialise the model to @c std::string.
+     *
+     * Serialise the given @p model to a std::string.
+     *
+     * @param model The model to serialise.
+     *
+     * @return The @c std::string representation of the model.
+     */
+    std::string printModel(ModelPtr model) const;
+
+    /**
+     * @brief Serialise the model to @c std::string.
+     *
+     * Serialise the given @p model to a std::string.
+     *
+     * @overload
+     *
+     * @param model The model to serialise.
+     *
+     * @return The @c std::string representation of the model.
+     */
+    std::string printModel(Model model) const;
```

Comment:
I did think about that but thought being consistent with parser was more important.  The Parser only takes strings so we can't overload on that.  Thoughts.